torchtext/experimental/utils/samplers/seqlen_sampler.py

```diff
@@ -0,0 +1,73 @@
+from typing import List
+
+import numpy as np
+from torch.utils.data import Dataset, IterableDataset
+from torch.utils.data.sampler import RandomSampler, Sampler, SequentialSampler
+
+
+class BucketByLengthBatchSampler(Sampler):
+    """Defines a batch sampler that batches examples of similar lengths together and
+    minimizes amount of padding needed.
+    Arguments:
+        data_source: data source to sample from.
+        bucket_boundaries: upper length boundaries to merge sentences with length
```

Comment:
How does this parameter work for batching examples according to the length?

Reply:
I have added more documentation on the docstring. Basically I'm borrowing the idea from tensorflow's `bucket_by_sequence_length`. The way it works is by categorizing raw data by putting them in a bucket whose lengths are in the upperbound range ``bucket_boundaries``. For ``bucket_boundaries`` = [5, 10], there will be three different buckets that will consist of sentences whose lengths are less than 5, between 5 and 10, and more than 10.